p2p/receiver/validator.go

```diff
@@ -101,6 +105,25 @@ func HandshakeValidator(networkID uint32, idService identity.Service) ValidatorG
 	}
 }
 
+func AccessValidator(ctx context.Context, doc *coredocumentpb.CoreDocument,  docReq *p2ppb.GetDocumentRequest, requesterCentId identity.CentID) error {
+
+	//checks which access type is relevant for the request
+	switch docReq.AccessType.String() {
+	case "ACCESS_TYPE_REQUESTER_VERIFICATION":
```

Comment:
Can you use the AccessType defined in the proto generated file instead of using the string representation?